enhancements/20220531-storage-tiering.md

```diff
@@ -0,0 +1,142 @@
+
+Users can access [Embedded Rancher UI](https://docs.harvesterhci.io/v1.0/troubleshooting/harvester/#access-embedded-rancher) to manage storageClass, or we can add Harvester's `StorageClass` page.
+
+##### 3.1 Create Volume
+Select storageClass (default to longhorn).
+
++ `Volumes` > `Create`（source=New), backing image storageClass need to be filtered out.
+![image](./20220531-storage-tiering/create-volume.png)
++ `Virtual Machines` > `Create` > `Volumes` > `Add Volume`, backing image storageClass need to be filtered out.
+![image](./20220531-storage-tiering/vm-add-volume.png)
++ `Snapshots` > snapshot > `Restore`
+  - For snapshot of an image volume, if source storageClass is not found, it can not be restored.
+  - For snapshot of other volume, only show storageClasses those have the same provisioner of source storageClass and choose source storageClassName by default. If source storageClass is not found, need show a warning message.
+  - For the above two points, save source provisioner, storageClassName,ImageID in snapshot's annotations during volume snapshot.
+
+##### 3.2 Show Volume
+```
+if provisioner is driver.longhorn.io
+  show diskSelector and nodeSelector 
+else
+  show storageClassName
+fi
+```
+
++ `Volumes` > volume
+![image](./20220531-storage-tiering/show-volume.png)
+
+#### 4 Image
+Since the storageClass of backing image is created dynamically, we only need to let users select diskSelector and nodeSelector during vm image creating.
+
+##### 4.1 Create Image
+
+Select diskSelector,nodeSelector (default to `any available`)
++ `Images` > `Create`
+  ![image](./20220531-storage-tiering/create-image.png)
++ `Volumes` > volume > `Export`
+  ![image](./20220531-storage-tiering/volume-export.png)
+
+##### 4.2 Show Image
+Show diskSelector,nodeSelector, if it exits
+
++ `Images` > image
+  ![image](./20220531-storage-tiering/show-image.png)
+
+
+### API changes
+
+Add a new field: `extraStorageClassParameters`, type `map[string]string`, to `harvesterhci.io.virtualmachineimages.spec`.
+
+If `extraStorageClassParameters` are allowed to be modified, the backend deletion and reconstruction of the storageClass being used may cause unpredictable consequences.
+And it is difficult to know from the storageClass information what the parameters were when the PVC was created.
+This makes displaying diskSelector and nodeSelector on the UI potentially inaccurate.
+Currently, the URL of image is also not allowed to be modified by the user on the UI
+So users should not be allowed to modify the `extraStorageClassParameters` contents.
+For different scheduling rules for the same image file, users need to create different images.
+
+```yaml
+spec:
+  checksum: ""
+  displayName: ubuntu-20.04-minimal-cloudimg-amd64.img
+  extraStorageClassParameters:
+    nodeSelector: "large"
+    diskSelector: "hdd"
+    migratable: "true"
+    numberOfReplicas: "3"
+    staleReplicaTimeout: "30"
+```
```

Comment:
are all those parameters allowed to be configured by the admin users?

Reply:
after updated according to https://github.com/harvester/harvester/pull/2335#issuecomment-1216292744
Yes